packages/store/addon/-private/system/references/reference.ts

```diff
@@ -29,7 +32,15 @@ function isResourceIdentiferWithRelatedLinks(
   return value && value.links && value.links.related;
 }
 
-export const INTERNAL_MODELS = new WeakMap<Reference, InternalModel>();
+export const REFERENCE_CACHE = new WeakMap<Reference, InternalModel | StableRecordIdentifier>();
```

Comment:
Once RECORD_ARRAY_MANAGER_IDENTIFIERS are turned on, lets make sure to simplify this, maybe leave a todo comment and rename to be more descriptive